apps/api/src/music/music.controller.ts

```diff
@@ -32,44 +34,109 @@ export class MusicController {
+        id: result.id,
+      },
+    };
+  }
+
+  @Post(':id/upload')
+  @ApiOperation({ description: 'Upload audio file for a specific music' })
   @UseInterceptors(
-    FileInterceptor('file', {
+    FileInterceptor('musicAudioFileName', {
       storage: diskStorage({
-        destination: multerDiskStorageDestination,
+        destination: multerDiskStorageMusicAudioFileDestination,
         filename: multerDiskStorageFilename,
       }),
       fileFilter: audioFileFilter,
     })
   )
-  async create(
-    @Body() data: CreateMusicDto,
+  async uploadMusic(
+    @Param('id') id: string,
     @UploadedFile()
-    file: Express.Multer.File
+    musicAudioFileName: Express.Multer.File
   ) {
-    if (!file) {
+    if (!musicAudioFileName) {
+      this.logger.log(musicAudioFileName);
       throw new BadRequestException('No file uploaded');
     }
 
-    const result = await this.musicService.create({
-      ...data,
-      fileName: file.filename,
+    const result = await this.musicService.update(id, {
+      musicAudioFileName: musicAudioFileName.filename,
     });
+
     this.logger.debug(`MUSIC CREATE | Recieved data client:\n${result}`);
-    this.logger.verbose(`MUSIC CREATE | Music file uploaded: ${file.filename}`);
+    this.logger.verbose(
+      `MUSIC CREATE | Music file uploaded: ${musicAudioFileName.filename}`
+    );
     return {
       message: 'Music was successfully created',
       success: true,
       data: {
-        date: result.createdAt,
-        fileName: file.filename,
+        musicAudioFileName: musicAudioFileName.filename,
         id: result.id,
       },
     };
   }
 
-  @Get('file/:id')
-  public async getFile(@Param('id') id: string) {
-    return this.musicService.getMusicFile(id);
+  @Post(':id/cover/upload')
+  @ApiOperation({
+    description: 'Upload music cover image file for a specific music',
+  })
+  @UseInterceptors(
+    FileInterceptor('coverImageFileName', {
+      storage: diskStorage({
+        destination: multerDiskStorageMusicCoverFileImageDestination,
+        filename: multerDiskStorageFilename,
+      }),
+      fileFilter: imageFileFilter,
+    })
+  )
+  async uploadMusicCover(
+    @Param('id') id: string,
+    @UploadedFile()
+    coverImageFileName: Express.Multer.File
+  ) {
+    if (!coverImageFileName) {
+      throw new BadRequestException('No file uploaded');
+    }
+
+    const result = await this.musicService.update(id, {
+      coverImageFileName: coverImageFileName.filename,
+    });
+
+    this.logger.debug(`MUSIC CREATE | Recieved data client:\n${result}`);
+    this.logger.verbose(
+      `MUSIC CREATE | Music file uploaded: ${coverImageFileName.filename}`
+    );
+    return {
+      message: 'Music was successfully created',
+      success: true,
+      data: {
+        coverImageFileName: coverImageFileName.filename,
+        id: result.id,
+      },
+    };
+  }
+
+  @Get(':id/file')
+  public async getMusicFile(@Param('id') id: string) {
```

Comment:
create an issue (tech-debt) to consider a CDN or some caching mechanism after measuring the performance of this endpoint (most likely we're gonna have performance issues, but we need to measure it first)

Reply:
You can follow this issue from here #21. This is some good stuff to consider :D